pkg/ring/kv/memberlist/tcp_transport.go

```diff
@@ -129,10 +147,20 @@ func NewTCPTransport(config TCPTransportConfig, logger log.Logger) (*TCPTranspor
 		ip := net.ParseIP(addr)
 
 		tcpAddr := &net.TCPAddr{IP: ip, Port: port}
-		tcpLn, err := net.ListenTCP("tcp", tcpAddr)
-		if err != nil {
-			return nil, fmt.Errorf("failed to start TCP listener on %q port %d: %v", addr, port, err)
+
+		var tcpLn net.Listener
+		if config.TLSEnabled {
+			tcpLn, err = tls.Listen("tcp", tcpAddr.String(), t.tlsConfig)
+			if err != nil {
+				return nil, fmt.Errorf("failed to start tls TCP listener on %q port %d: %v", addr, port, err)
```

Comment:
Should these be `%w` for the error?

Reply:
Yea, I think this would be a good fit for that formatting directive.

Reply:
I agree on `%w` but, doing a step forward, what if we move to `errors.Wrapf()` like we're currently doing everywhere? This looks material for a follow up PR, not to be done in this one.

Reply:
👍

Reply:
`%w/%v` makes no difference, until we want to unwrap the errors and work with them. If we just log them, it doesn't matter.